index.d.ts

```diff
@@ -22,7 +22,7 @@ declare interface limitInterface<T> {
    * @param items any items
    * @param mapper iterator
    */
-  map(items: [T], mapper: (T) => Promise<T>): Promise<[T]>
+  map(items: T[], mapper: (T) => Promise<T>): Promise<[T]>
```

Comment:
Additionally, the mapper function's type should be `mapper: (item: T) => Promise<T>` (in the current code, `T` is parsed as a parameter name)